models/db.go

```diff
@@ -5,21 +5,47 @@ import (
 	"github.com/boltdb/bolt"
 )
 
+/**
+ * - interface defining operations of our 'notes' datastore
+ * - notice the function prototypes here have one-to-one mapping with
+ *   *almost* every command (plus it's subcommand / options)
+ */
 type Datastore interface {
+	// notebook-related operations
 	AddNotebook(notebook Notebook) error
 	GetNotebook(notebookTitle string) (Notebook, error)
-	GetAllNotebooks() ([]Notebook, error)
+	GetAllNotebooks(onlyNames bool) ([]Notebook, error)
```

Comment:
It would be better if the API had separate methods for this call, eg:
```GetAllNotebooks``` and ```GetAllNotebookNames```.

Passing a boolean as the only argument to the function would be confusing for someone who is not familiar with the entire codebase.

Reply:
yes valid point.
 - i've reverted signature of `GetAllNotebooks`
 - and have added `GetAllNotebookNames`
 https://github.com/y2k-shubham/notes/pull/7/files#diff-c1ad6f79546387c376f9cade738991bbR17